app/helpers/statistics_announcement_helper.rb

```diff
@@ -0,0 +1,26 @@
+module StatisticsAnnouncementHelper
+  def on_in_between_for_release_date(date)
+    return "on #{date}" if date_is_exact_format?(date)
+    return "in #{date}" if date_is_one_month_format?(date)
+    return "between #{replace_on_with_and(date)}" if date_is_two_month_format?(date)
+  end
+
+private
+
+  def replace_on_with_and(date_in_two_month_format)
+    re = /\s(to)\s/
+    date_in_two_month_format.sub(re, " and ")
+  end
+
+  def date_is_two_month_format?(date)
+    date =~ /\A(\w+)\s(to)\s(\w+)/
+  end
+
+  def date_is_one_month_format?(date)
+    date =~ /\A(\w+)\s(\d{1,4})/
+  end
+
+  def date_is_exact_format?(date)
+    date.downcase =~ /\A(\d+)\s(\w+)\s(\d{4})\s(\d{1,2}:\d{1,2})(am|pm)/
```

Comment:
That's some funky regexin’

Reply:
:) yeah anyone with better regex-fu who fancies neatening these up gets :1st_place_medal:  from me.  I'm slightly worried there's an edge case I haven't thought of that fails the regex.

Reply:
Do you only care at this point whether it's a valid date? you might be able to skip this regex and just do Date.parse or a similar method at this point

Reply:
The problem I had with `Date.parse` is that it seems very forgiving:

```
irb(main):001:0> Date.parse("November to December 2017")
=> Fri, 01 Dec 2017
irb(main):002:0> Date.parse("January 2018")
=> Mon, 01 Jan 2018
irb(main):003:0> Date.parse("13 December 2017 9:30am")
=> Wed, 13 Dec 2017
```

so I can't rely on it to determine the format of the string since they are all valid dates.

Reply:
Yup sure, though you'll have already checked for the first two scenarios with your simpler regex so it'd only be for dealing with the last of those scenarios.

Reply:
so having thought about this a bit and reading through the rest of the comments around the issue, I noticed the code didn't take into account a completely unexpected format being passed in. If some string got passed in here that failed all three conditions, we would get a front end message that said "These statistics will be released". I've updated to return any unexpected format back so we at least append the date to the sentence as we do now without an "in", "on" or "between".

To take that into account, I think I still need to check for this format specifically - what do you think @kevindew?

Reply:
good catch 👍 

I don't think it's necessary to check that one specifically, but it's more in the realm of two ways to skin a cat than offering a massive advantage to use `Date.Parse`. Only seems like something worth doing if there are risks the formatting could be slightly different - which seems unlikely.